ablySpec/RestClient.swift

```diff
@@ -198,6 +200,193 @@ class RestClient: QuickSpec {
+                                    done()
+                                }
+                                // Different token
+                                expect(tokenDetailsA?.token).toNot(equal(tokenDetailsB?.token))
+                                done()
+                            }
+                        }
+                    }
+                }
+            }
+
+            // RSC14b
+            context("basic authentication flag") {
+                it("should be true when key is set") {
+                    let client = ARTRest(key: "key:secret")
+                    expect(client.auth.options.isBasicAuth()).to(beTrue())
+                }
+
+                for (caseName, caseSetter) in AblyTests.authTokenCases {
+                    it("should be false when \(caseName) is set") {
+                        let options = ARTClientOptions()
+                        caseSetter(options)
+
+                        let client = ARTRest(options: options)
+
+                        expect(client.auth.options.isBasicAuth()).to(beFalse())
+                    }
+                }
+            }
+
+            // RSC14c
+            it("should error when expired token and no means to renew") {
+                let client = ARTRest(options: AblyTests.commonAppSetup())
+                let auth = client.auth
+
+                let tokenParams = ARTAuthTokenParams()
+                tokenParams.ttl = 3.0 //Seconds
+
+                waitUntil(timeout: testTimeout) { done in
+                    auth.requestToken(tokenParams, withOptions: nil) { tokenDetails, error in
+                        if let e = error {
+                            XCTFail(e.description)
+                            done()
+                        }
+                        else if let currentTokenDetails = tokenDetails {
+                            let options = AblyTests.clientOptions()
+                            options.key = client.options.key
+
+                            // Expired token
+                            options.tokenDetails = ARTAuthTokenDetails(token: currentTokenDetails.token, expires: currentTokenDetails.expires?.dateByAddingTimeInterval(testTimeout), issued: currentTokenDetails.issued, capability: currentTokenDetails.capability, clientId: currentTokenDetails.clientId)
+
+                            options.authUrl = NSURL(string: "http://test-auth.ably.io")
+
+                            let rest = ARTRest(options: options)
+                            rest.httpExecutor = mockExecutor
+
+                            // Delay for token expiration
+                            delay(tokenParams.ttl) {
+                                // 40140 - token expired and will not recover because authUrl is invalid
+                                publishTestMessage(rest) { error in
+                                    expect(mockExecutor.responses.first?.allHeaderFields["X-Ably-ErrorCode"] as? String).to(equal("40140"))
```

Comment:
Ignore previous comment about checking 40140 as I can see this was introduced in a later commit.  However, the error object emitted should contain the error code and message, so I think it would make sense to check that the error object itself contains this information.  This is needed for a developer who potentially catches the error and needs to understand why there was a failure.